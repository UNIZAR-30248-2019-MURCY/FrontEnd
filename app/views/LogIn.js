import React, { Component } from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  Image,
  Text
} from 'react-native'
import { Button } from 'react-native-elements';
import { Link } from './../config/routing';

export default class LogIn extends Component {
  state = {
    username: '', password: '', email: '', phone_number: ''
  }
  onChangeText = (key, val) => {
    this.setState({ [key]: val })
  }
  logIn = async () => {
    const { email, password } = this.state
    try {
      
      // here place your signup logic
      //console.log('user successfully signed up!: ', success)
    } catch (err) {
      //console.log('error signing up: ', err)
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <Image
          style={styles.logo}
          source={require('./../assets/images/murcy.png')}
        />
        <TextInput
          style={styles.input}
          placeholder='Email'
          autoCapitalize="none"
          placeholderTextColor='darkgrey'
          onChangeText={val => this.onChangeText('email', val)}
        />
        <TextInput
          style={styles.input}
          placeholder='Password'
          secureTextEntry={true}
          autoCapitalize="none"
          placeholderTextColor='darkgrey'
          onChangeText={val => this.onChangeText('password', val)}
        />

        <Button
          buttonStyle={styles.button}
          title="Log In"
          onPress={() => {this.logIn}} />

        <Link to="/signin">
          <Text>Sign Up</Text>
        </Link>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  input: {
    width: 350,
    height: 55,
    backgroundColor: 'whitesmoke',
    margin: 10,
    padding: 8,
    color: 'grey',
    borderRadius: 14,
    fontSize: 18,
    fontWeight: '500',
  },
  container: {
    flex: 1,
    alignItems: 'center',
    //justifyContent: 'center',
    marginTop: 100,
    padding: 20
  },
  logo: {
    margin: 30,
    padding: 8,
    width: 100,
    height: 100
  },
  signUpScreenButton: {
    marginRight: 40,
    marginLeft: 40,
    marginTop: 10,
    paddingTop: 10,
    paddingBottom: 10,
    backgroundColor: 'grey',
    borderRadius: 10,
  },
  button: {
    width: 150,
    height: 55,
    margin: 40,
    padding: 8,
    backgroundColor: 'grey',
    borderRadius: 14
  }

})