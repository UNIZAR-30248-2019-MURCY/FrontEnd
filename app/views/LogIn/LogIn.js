import React, {Component} from 'react';
import {
    View,
    TextInput,
    StyleSheet,
    Image,
    Text,
    ActivityIndicator
} from 'react-native'
import {Button} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import {logInUser, userInfo} from "../../services/user/userFuncs";
import {saveData} from "../../services/AsyncStorage/save";
import {NavigationActions, StackActions} from "react-navigation";


export default class LogIn extends Component {

    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            token: '',
            error: false,
            userData: '',
            loading: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if (this.state.username !== '' && this.state.password !== '') {
            this.setState({loading: true})
            this.setState({error: false})
            logInUser(this.state.username, this.state.password)
                .then((data) => {
                    saveData('token', data).then(r =>
                        userInfo(data.jsonWebToken).then((dataUser) => {
                                saveData('role', dataUser.role).then(r =>
                                        console.log(dataUser.role),
                                    this.setState({loading: false}),
                                    this.props.navigation.navigate('App')
                                )
                            }
                        )
                            .catch((error) => {
                                this.setState({error: error.message}),
                                    this.setState({loading: false})
                            })
                    );
                })
                .catch((error) => {
                    this.setState({error: error.message}),
                        this.setState({loading: false})
                });
        } else {
            this.setState({error: 'Introduzca todos los campos'})
        }
    }

    render() {
        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]}>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        return (

            <View style={styles.container}>
                <View style={styles.cross}>
                    <Button
                        className='close-button'
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.navigation.navigate('AuthLoading')
                        }
                        }
                    />
                </View>

                <View style={styles.login}>
                    <Image
                        style={styles.logo}
                        source={require('../../assets/images/murcy.png')}
                    />
                    <TextInput
                        className='userInput'
                        style={styles.input}
                        placeholder='Username'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('username', val)}
                    />
                    <TextInput
                        className='passInput'
                        style={styles.input}
                        placeholder='Password'
                        secureTextEntry={true}
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onSubmitEditing={() => {
                            this.handleSubmit()
                        }}
                        onChangeText={val => this.onChangeText('password', val)}
                    />
                    {showErr}
                    {showLoading}
                    <Button
                        className='login-button'
                        buttonStyle={styles.button}
                        title="Log In"
                        onPress={() => {
                            this.handleSubmit()
                        }}/>
                    <Button
                        className='signup-button'
                        type="clear"
                        buttonStyle={styles.button2}
                        title="Sign Up"
                        titleStyle={{color: 'grey'}}
                        onPress={() => {
                            this.props.navigation.replace('SignUp');
                        }
                        }/>
                </View>

            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    containerLoading: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10
    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    login: {
        alignItems: 'center',
    },
    logo: {
        marginBottom: 30,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    error: {
        margin: 10,

    },

})
