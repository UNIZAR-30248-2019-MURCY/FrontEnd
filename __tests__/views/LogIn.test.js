import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import LogIn from "../../app/views/LogIn";

describe('<LogIn />', () => {
    it('LogIn renders without crashing', () => {
        const rendered = renderer.create(<LogIn />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('LogIn test against snapshot', () => {
        const welcomePage = renderer.create(<LogIn />).toJSON();
        expect(welcomePage).toMatchSnapshot();
    });
    it('should navigate to Sign Up', () => {
        const historyMock = { replace: jest.fn() };
        const wrapper = shallow(<LogIn history={historyMock} />);
        wrapper.find('.signup-button').simulate('press');
        expect(historyMock.replace.mock.calls.length).toEqual(1);
        expect(historyMock.replace.mock.calls[0]).toEqual(['/signup']);
    })
});
