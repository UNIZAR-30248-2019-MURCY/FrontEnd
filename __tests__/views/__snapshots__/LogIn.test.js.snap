// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn /> LogIn test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginBottom": 40,
        "marginRight": 10,
        "marginTop": 50,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "marginBottom": 30,
          "padding": 8,
          "width": 100,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="darkgrey"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "whitesmoke",
          "borderRadius": 14,
          "color": "grey",
          "fontSize": 18,
          "fontWeight": "500",
          "height": 55,
          "margin": 10,
          "padding": 8,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="darkgrey"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "whitesmoke",
          "borderRadius": 14,
          "color": "grey",
          "fontSize": 18,
          "fontWeight": "500",
          "height": 55,
          "margin": 10,
          "padding": 8,
          "width": 300,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View />
    <View
      style={
        Object {
          "borderRadius": 14,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "grey",
              "borderColor": "#2089dc",
              "borderRadius": 14,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 55,
              "justifyContent": "center",
              "margin": 10,
              "marginTop": 20,
              "padding": 8,
              "width": 150,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Log In
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRadius": 14,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 14,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 55,
              "justifyContent": "center",
              "padding": 8,
              "width": 150,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
